### PR TITLE
Add recipe for axel downloader.

### PR DESCRIPTION
**Pull request by jdblischak**
[axel](https://github.com/axel-download-accelerator/axel) is a lightweight CLI download accelerator written in C.

Note that there is also a [PyPI package named axel](https://pypi.org/project/axel/) and some other [conda recipes for "axel" on Anaconda Cloud](https://anaconda.org/search?q=axel). Would it be better to give the recipe a more descriptive name like "axel-download-accelerator"?

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by jdblischak** [comment]
The [documentation](https://github.com/axel-download-accelerator/axel#dependencies) states that gettext is a dependency, but the [log](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=278210&view=logs&j=6f142865-96c3-535c-b7ea-873d86b887bd&t=22b0682d-ab9e-55d7-9c79-49f3c3ba4823&l=910) has a warning stating that it isn't used.

```
WARNING (axel): run-exports library package conda-forge::gettext-0.19.8.1-h0b5b191_1005 in requirements/run but it is not used (i.e. it is overdepending or perhaps statically linked? If that is what you want then add it to `build/ignore_run_exports`)
```

As far as I can tell, axel isn't vendoring gettext. My recipe is using the release tarball, but maybe I have to go back to the `autoconf` step to make sure gettext is properly linked ([documentation](https://github.com/axel-download-accelerator/axel#building-on-mac-os-x-homebrew)).

I'd appreciate any advice.

**Comment by jdblischak** [comment]
@conda-forge/help-c-cpp Could you please advise on the dependencies for this recipe? The [documentation](https://github.com/axel-download-accelerator/axel#dependencies) states that it depends on gettext, pkg-config, and libssl. I listed gettext in host. Should I move it to build?

**Comment by jdblischak** [comment]
I downloaded the [artifacts](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=278210&view=artifacts&pathAsName=false&type=publishedArtifacts) for both Linux and macOS to confirm they work. Here's example code for macOS:

```zsh
mv ~/Downloads/conda_pkgs_osx.zip /tmp
cd /tmp
mamba create -n test-axel python
conda activate test-axel
unzip conda_pkgs_osx.zip
conda install conda_pkgs_osx/axel-2.17.10-hcd1c8dc_0.tar.bz2
axel --version
axel --output=github.html https://github.com/
```

**Comment by jdblischak** [comment]
@conda-forge/staged-recipes Please review. I confirmed this recipe generates functional binaries, but I have some questions about best practices:

1. Are the dependencies correct? How can I remove the warning about gettext?
2. Should I keep the recipe name as "axel" or make it more specific in order to avoid future name clashes?

File: recipes/axel/build.sh [hunk 1]
```diff
@@ -0,0 +1,5 @@
+#!/bin/bash
+
+./configure --disable-Werror --prefix=$PREFIX
+make
+make install
```
Comment (ocefpaf) [inline recipes/axel/build.sh:5]: Is there a make check?

**Comment by ocefpaf** [comment]
> 1. Are the dependencies correct? How can I remove the warning about gettext?

Not sure but running more test, if they exist, would help.

> 2. Should I keep the recipe name as "axel" or make it more specific in order to avoid future name clashes?

Some names are on a first come first served, so I don't think you need to change it.

File: recipes/axel/build.sh (hunk 1 above)
Comment (jdblischak) [reply recipes/axel/build.sh:5]: The only reference to `make check` I could find was in the generic documentation, so I'm not sure if they've implemented it. Worth a try.

https://github.com/axel-download-accelerator/axel/blob/bd8b313d2e8b2dcf4d934eaa2985424f961112b9/INSTALL#L61

**Comment by jdblischak** [comment]
I've added additional tests, including an explicit file download, and they all pass. Thus the binary produced appears to be functional. I'm happy with this recipe, so unless there are objections, please merge. Thanks!

**Pull request merged**